### PR TITLE
[Network] support host names in application gateway listener

### LINKED ISSUE #11743: CLI support for Wildcard Listener in Application Gateway
**What is the Azure resource provider your feature is part of?**

NRP

**Provide a brief description of the feature or work requested. A link to conceptual documentation may be helpful too.**

CLI support is needed to add wildcard hostnames in a listener of Application Gateway. This will add an additional parameter 'host-names' in 'az network application-gateway http-listener' command - https://docs.microsoft.com/en-us/cli/azure/network/application-gateway/http-listener?view=azure-cli-latest. This parameter will be valid only for Application Gateway Standard_v2 and WAF_v2 SKUs. The validations in NRP are in place for the SKU verification.
This parameter takes values as an array of strings.

**What is the minimum API version of your service required to implement your feature?**

2019-09-01

**Provide a link to the location of your feature(s) in the REST API specs repo. If your feature(s) has corresponding commit or pull request in the REST API specs repo, provide them. This should be on the master branch of the REST API specs repo.**

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/applicationGateway.json

**If you have a target date for release of this feature/work, please provide it. While we can't guarantee these dates,
it will help us prioritize your request against other requests.**

End of January 2020. The API is already out and we are working on PS and Portal, once done, we'll be targeting Public Preview.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fix #11743
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -378,6 +380,9 @@ def update_ag_http_listener(cmd, instance, parent, item_name, frontend_ip=None,
         if firewall_policy is not None:
             instance.firewall_policy = SubResource(id=firewall_policy)
 
+    if host_names is not None:
+        instance.host_names = host_names or None
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/network/custom.py]: under the condition of `host_names` is not None, why need `or` in assignment?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: "" this one?

Comment (haroldrandom) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: I mean the default vaule of host_names is None, why test under the if statement ?

**Pull request merged**